sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -609,6 +610,19 @@ object CollapseWindow extends Rule[LogicalPlan] {
 }
 
 /**
+ * Transpose Adjacent Window Expressions.
+ * - If the partition spec of the parent Window expression is a subset of the partition spec
+ *   of the child window expression, transpose them.
+ */
+object TransposeWindow extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    case w1 @ Window(we1, ps1, os1, w2 @ Window(we2, ps2, os2, grandChild))
+        if ps1.length < ps2.length && ps2.containsSlice(ps1) =>
```

Comment:
You need to check that the windows are independent, e.g.: `w1.references.intersect(w2.windowOutputSet).isEmpty`